### PR TITLE
feat: config options for bloom filter cache

### LINKED ISSUE #8279: Feature: make bloom filter index cache configurable
**Summary**

make bloom filter index cache configurable

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

config options for bloom filter cache

- new config options for bloom index cache 
~~~
    /// Max number of cached bloom index meta objects
    #[clap(long, default_value = "3000")]
    pub table_cache_bloom_index_meta_count: u64,

    /// Max bytes of cached bloom index, default value is 1GB
    #[clap(long, default_value = "1073741824")]
    pub table_cache_bloom_index_data_bytes: u64,
~~~
- remove unused config options
 ~~~
    /// Max number of cached table block meta
    pub table_cache_block_meta_count: u64,
    /// Table memory cache size (mb)
    pub table_memory_cache_mb_size: u64,
    /// Table disk cache folder root
    pub table_disk_cache_root: String,
    /// Table disk cache size (mb)
    pub table_disk_cache_mb_size: u64,
 ~~~

Fixes #8279

### CODE REVIEW

File: src/query/config/src/outer_v0.rs [hunk 1]
```diff
@@ -894,21 +894,13 @@ pub struct QueryConfig {
     #[clap(long, default_value = "10240")]
     pub table_cache_segment_count: u64,
 
-    /// Max number of cached table block meta
-    #[clap(long, default_value = "102400")]
-    pub table_cache_block_meta_count: u64,
```
[the remark this answers is not in the document]
Comment (BohuTANG) [reply src/query/config/src/outer_v0.rs:899]: If these configs no need we can remove them, users may config them but no affected.

[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/query/config/src/outer_v0.rs:899]: > If these configs no need we can remove them, users may config them but no affected,

Removing them can't help resolve this problem.

We should keep those configs, print error messages, and return errors to users if we don't want users to use them anyway.

We can do those jobs while converting from outer config to inner config:

image

After all users have upgrade to this version, we can remove them without panic our user.

[the remark this answers is not in the document]
Comment (dantengsky) [reply src/query/config/src/outer_v0.rs:899]: they are somehow misleading... we doc them and users set them, but no effects.
keep them for backward compatibility?

[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/query/config/src/outer_v0.rs:899]: > we doc them and users set them, but no effects.

Those deprecated configs will be kept in code only for backward compatibility:

- Our docs will not mention them. 
- We should print warning or error message if user tried to set them.

[the remark this answers is not in the document]
Comment (dantengsky) [reply src/query/config/src/outer_v0.rs:899]: > We should keep those configs, print error messages, and return errors to users if we don't want users to use them anyway.

er... I must have missed something here.

I think for any version, 
`./databend-query --no-such-option xx` will print error message.

shall we make an announcement, and then remove those config options?

[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/query/config/src/outer_v0.rs:899]: > `./databend-query --no-such-option xx` will print error message.

If we kept those configs in `outer config`, we could print a helpful message and allow databend-query to continue to run.

[the remark this answers is not in the document]
Comment (dantengsky) [reply src/query/config/src/outer_v0.rs:899]: ok. let's follow the release roadmap :)

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**